### PR TITLE
options/posix: Add mqueue header

### PR DESCRIPTION
**Pull request by Dennisbonke**
Part of the mlibc LFS project.

### CODE REVIEW

File: options/posix/include/mqueue.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	long mq_msgsize;
+	long mq_curmsgs;
+	long __pad[4];
+};
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif // _MQUEUE_H
```
Comment (ArsenArsen) [inline options/posix/include/mqueue.h:21]: Don't add C++-style headers to public-facing interfaces.

**Review by ArsenArsen — CHANGES_REQUESTED** [review]

File: options/posix/include/mqueue.h [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct mq_attr {
+	long mq_flags;
+	long mq_maxmsg;
+	long mq_msgsize;
+	long mq_curmsgs;
+	long __pad[4];
```
[the remark this answers is not in the document]
Comment (Dennisbonke) [reply options/posix/include/mqueue.h:14]: Unfortunately I think you're right, linux has a bits/mqueue.h header which defines the struct. I'll move it over to abi-bits.

**Review by Geertiebear — APPROVED** [review]
Unfortunately this PR conflicts on latest master, rebase it and it's good to go.

File: sysdeps/linux/meson.build [hunk 3]
```diff
@@ -62,6 +62,8 @@ if not no_headers
 		'include/abi-bits/inotify.h',
 		'include/abi-bits/clockid_t.h',
 		'include/abi-bits/shm.h',
+		'include/abi-bits/mqueue.h',
+		host_machine.cpu_family() / 'abi-bits/arch-syscall.h',
```
Comment (64) [inline sysdeps/linux/meson.build:66]: looks unrelated

File: options/posix/include/mqueue.h [hunk 4]
```diff
@@ -0,0 +1,18 @@
+
```
Comment (64) [inline options/posix/include/mqueue.h:1]: why is there a newline :^)

**Review by 64 — CHANGES_REQUESTED** [review]

File: options/posix/include/mqueue.h (hunk 4 above)
Comment (ArsenArsen) [reply options/posix/include/mqueue.h:1]: in other places also

File: sysdeps/linux/meson.build (hunk 3 above)
Comment (ArsenArsen) [reply sysdeps/linux/meson.build:66]: this was removed yes

**Review by no92 — APPROVED** [review]

**Review by 64 — APPROVED** [review]

**Review by Geertiebear — APPROVED** [review]

**Pull request merged**